colab/tflite_text_classification.ipynb

```diff
@@ -0,0 +1,514 @@
+  "metadata": {
+    "colab": {
+      "name": "Copy of tflite_text_classification.ipynb",
+      "provenance": [],
+      "collapsed_sections": [
+        "dweksnWs1eZb"
+      ]
+    },
+    "kernelspec": {
+      "name": "python3",
+      "display_name": "Python 3"
+    },
+    "language_info": {
+      "name": "python"
+    }
+  },
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "_VsBo9D7GJ8G"
+      },
+      "source": [
+        "```\n",
+        "Copyright 2020 The IREE Authors\n",
+        "\n",
+        "Licensed under the Apache License v2.0 with LLVM Exceptions.\n",
+        "See https://llvm.org/LICENSE.txt for license information.\n",
+        "SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception\n",
+        "```"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "3Mjqf_WmSkAs"
+      },
+      "source": [
+        "# TFLite text classification sample with IREE\n",
+        "\n",
+        "This notebook demonstrates how to download, compile, and run a TFLite model with IREE.  It looks at the pretrained [text classification](https://www.tensorflow.org/lite/examples/text_classification/overview) model, and shows how to run it with both TFLite and IREE.  The model predicts if a sentence's sentiment is positive or negative, and is trained on a database of IMDB movie reviews.\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "W71slxNjS3SB"
+      },
+      "source": [
+        "# Setup"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "T8WMt2sPS4ft"
+      },
+      "source": [
+        "%%capture\n",
+        "!python -m pip install iree-compiler-snapshot iree-runtime-snapshot iree-tools-tflite-snapshot -f https://github.com/google/iree/releases/snapshot-20210521.291\n",
+        "!pip3 install --extra-index-url https://google-coral.github.io/py-repo/ tflite_runtime"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "cellView": "code",
+        "id": "7L4_gnRVVBdi"
+      },
+      "source": [
+        "import numpy as np\n",
+        "import urllib.request\n",
+        "import pathlib\n",
+        "import tempfile\n",
+        "import re\n",
+        "import tflite_runtime.interpreter as tflite\n",
+        "\n",
+        "from iree import runtime as iree_rt\n",
+        "from iree.compiler import compile_str\n",
+        "from iree.tools import tflite as iree_tflite\n",
+        "\n",
+        "ARTIFACTS_DIR = pathlib.Path(tempfile.gettempdir(), \"iree\", \"colab_artifacts\")\n",
+        "ARTIFACTS_DIR.mkdir(parents=True, exist_ok=True)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "dweksnWs1eZb"
+      },
+      "source": [
+        "## Load the TFLite model"
```

Comment:
Since this is collapsed by default now, you could put more information in this text cell to answer "should I look at these details?"

![image](https://user-images.githubusercontent.com/4010439/120378984-f6d56400-c2d3-11eb-97b5-e1024d15e0a0.png)

Copying my earlier message:

> There should be enough context there that people can see if their own different models will need similar treatment, or if they can skip over that section entirely.

Suggested wording (feel free to go with something else):

```
## Load the TFLite model

* Download files for the pretrained model
* Extract model metadata used for input pre-processing and output post-processing
* Define helper functions for pre- and post-processing

These steps will differ from model to model. Consult the model source or reference documentation for details.
```